### PR TITLE
[Merged by Bors] - feat(Algebra/Homology): the canonical truncation `truncGE`

### PR DESCRIPTION
**Pull request by joelriou**
We construct the canonical truncation `truncGE` of homological complexes relative to an embedding `e` of complex shapes which satisfies `e.IsTruncGE`.

---

- [x] depends on: #14650

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/Embedding/Boundary.lean [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+only element in this lower boundary. Similarly, we define
+`Embedding.BoundaryLE`.
+
+-/
+
+namespace ComplexShape
+
+namespace Embedding
+
+variable {ι ι' : Type*} (c : ComplexShape ι) (c' : ComplexShape ι') (e : Embedding c c')
```
Comment (erdOne) [inline Mathlib/Algebra/Homology/Embedding/Boundary.lean:39]:
```suggestion
variable {ι ι' : Type*} {c : ComplexShape ι} {c' : ComplexShape ι'} (e : Embedding c c')
```

File: Mathlib/Algebra/Homology/Embedding/Boundary.lean [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    apply c'.prev_eq_self
+    intro hj'
+    simp only [BoundaryGE, not_and, not_forall, not_not] at hj
+    obtain ⟨i, hi⟩ := hj hj'
+    rw [e.rel_iff] at hi
+    rw [c.prev_eq' hi] at hij
+    exact hij' (by simpa only [hij] using hi)
+
+variable {e} in
+lemma BoundaryGE.false {j : ι} (hj : e.BoundaryGE j) [e.IsTruncLE] : False := by
```
Comment (erdOne) [inline Mathlib/Algebra/Homology/Embedding/Boundary.lean:90]: should `isTruncLE` be in the name?

Comment (joelriou) [reply Mathlib/Algebra/Homology/Embedding/Boundary.lean]: Thanks! It is indeed better to include it in the name.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**